### PR TITLE
D3 graph setup #79

### PR DESCRIPTION
**Pull request by Mehrdadmaskull**

### CODE REVIEW

File: Client/src/Components/Graph/Graph.js [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+import * as d3 from "d3"
+
+//Global variables for graph creation
+
+var margin = {
+    top: 50,
+    right: 210,
+    bottom: 50,
+    left: 70
+},
```
Comment (GD-Dheer) [inline Client/src/Components/Graph/Graph.js:11]: I think we should be using consts instead of vars

File: Client/src/Components/Graph/Graph.js [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    outerWidth = 1050,
+    outerHeight = 500,
+    width = outerWidth - margin.left - margin.right,
+    height = outerHeight - margin.top - margin.bottom;
+
+var xScale, yScale, xAxis, yAxis;
+
+//This function creates the basis for the linear x and y axis
+function startLinear() {
```
Comment (GD-Dheer) [inline Client/src/Components/Graph/Graph.js]: Similarly const function = () =>

Arrow functions are the way to go!

File: Client/src/Components/Graph/Graph.js [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+var dataset4 = [{ x: 3, y: 7 }, { x: 1, y: 9 }, { x: 8, y: 7 }, { x: 1, y: 4 }, { x: 8, y: 5 }]
+//Datalist is the list fed to the graphCreation
+var datalist = []
+var colourslist = []
+datalist.push(dataset1)
+datalist.push(dataset2)
+datalist.push(dataset3)
+datalist.push(dataset4)
+
+class Graph extends Component {
```
Comment (GD-Dheer) [inline Client/src/Components/Graph/Graph.js:53]: Class components are also old-age... Functional components all the way. This lets us use Hooks

**Review by GD-Dheer — CHANGES_REQUESTED** [review]
Would be A1 if you could add changes early... These wont take long

**Review by GD-Dheer — APPROVED** [review]

**Review by GD-Dheer — APPROVED** [review]

File: Client/src/Components/Graph/Graph.js (hunk 1 above)
Comment (TiraFire) [reply Client/src/Components/Graph/Graph.js:11]: The margins, outers and inners are changed to const, but the xscale and axes will change will scaling so it is left as a var

File: Client/src/Components/Graph/Graph.js (hunk 2 above)
Comment (TiraFire) [reply Client/src/Components/Graph/Graph.js]: done!

File: Client/src/Components/Graph/Graph.js (hunk 3 above)
Comment (TiraFire) [reply Client/src/Components/Graph/Graph.js:53]: Working on it, will create another issue for this problem

**Review by TiraFire — COMMENTED** [review]
Im confused, why did you remove the title and div?

**Review by ratulkai93 — APPROVED** [review]
Graph is working fine. good work!

**Pull request merged**